### DISCUSSION TITLE
[API v3 PRODUCTION BUG] findAllActiveListingsByShop sort_on "updated" is not working

### DISCUSSION THREAD

**Discussion by adelobosko** [etsy/open-api#778] [Q&A]
[findAllActiveListingsByShop](https://developers.etsy.com/documentation/reference/#operation/findAllActiveListingsByShop) endpoint has problems.

sort_on **updated** is not working, result is sorts by **created**
GET  https://openapi.etsy.com/v3/application/shops/35757357/listings/active?limit=25&offset=0&sort_on=updated&sort_order=desc

Response:
```
{
  "count": 182,
  "results": [
    {
      "listing_id": 1376936721,
      "user_id": 639704972,
      "shop_id": 35757357,
      "creation_timestamp": 1671702628,
      "created_timestamp": 1671702628,
      "ending_timestamp": 1682153428,
      "original_creation_timestamp": 1671702628,
      "last_modified_timestamp": 1671770892,
      "updated_timestamp": 1671770892,
	  ...,
      "state_timestamp": 1671702635,
    },
    {
      "listing_id": 1376171709,
      "user_id": 639704972,
      "shop_id": 35757357,
      "creation_timestamp": 1671619226,
      "created_timestamp": 1671619226,
      "ending_timestamp": 1682070026,
      "original_creation_timestamp": 1671619226,
      "last_modified_timestamp": 1671684496,
      "updated_timestamp": 1671684496,
	  ...,
      "state_timestamp": 1671619226,
    },
    {
      "listing_id": 1357614998,
      "user_id": 639704972,
      "shop_id": 35757357,
      "creation_timestamp": 1671106880,
      "created_timestamp": 1671106880,
      "ending_timestamp": 1681557680,
      "original_creation_timestamp": 1671106880,
      "last_modified_timestamp": 1671526523,
      "updated_timestamp": 1671526523,
	  ...,
      "state_timestamp": 1671106880,
    },
    {
      "listing_id": 1357541834,
      "user_id": 639704972,
      "shop_id": 35757357,
      "creation_timestamp": 1671095195,
      "created_timestamp": 1671095195,
      "ending_timestamp": 1681545995,
      "original_creation_timestamp": 1671095195,
      "last_modified_timestamp": 1671796097,
      "updated_timestamp": 1671796097,
	  ...,
      "state_timestamp": 1671095195,
    },
    {
      "listing_id": 1355284600,
      "user_id": 639704972,
      "shop_id": 35757357,
      "state": "active",
      "creation_timestamp": 1670872059,
      "created_timestamp": 1670872059,
      "ending_timestamp": 1681322859,
      "original_creation_timestamp": 1670872059,
      "last_modified_timestamp": 1671526791,
      "updated_timestamp": 1671526791,
	  ...,
      "state_timestamp": 1670872059,
    },
    ...,
  ]
}
```

| index | creation_timestamp | [n] >= [n+1] |
| --- | ------------------------ | --- |
|  0  | 1671702628                 |      |
|  1  | 1671619226                 |  +  |
|  2  | 1671106880                 |  +  |
|  3  | 1671095195                 |  +  |
|  4  | 1670872059                 |  +  |

| index | last_modified_timestamp | [n] >= [n+1] |
| --- | ------------------------------ | -------------- |
|  0  | 1671770892                         |                       |
|  1  | 1671684496                         |  -                   |
|  2  | 1671526523                         |  -                   |
|  3  | 1671796097                         |  +                  |
|  4  | 1671526791                         |  -                   |

| index | updated_timestamp | [n] >= [n+1] |
| --- | ------------------------- | -------------- |
|  0  | 1671770892                  |                       |
|  1  | 1671684496                  |  -                   |
|  2  | 1671526523                  |  -                   |
|  3  | 1671796097                  |  +                  |
|  4  | 1671526791                  |  -                   |

Same with **&keywords=Liquida**
GET  https://openapi.etsy.com/v3/application/shops/35757357/listings/active?limit=25&offset=0&sort_on=updated&sort_order=desc&keywords=Liquida

Response:
| index | creation_timestamp | [n] >= [n+1] |
| --- | --- | --- |
| 0 | 1671702628 |  |
| 1 | 1671619226 | + |
| 2 | 1671106880 | + |
| 3 | 1670872059 | + |
| 4 | 1670872043 | + |

| index | last_modified_timestamp | [n] >= [n+1] |
| --- | --- | --- |
| 0 | 1671770892 | |
| 1 | 1671684496 | + |
| 2 | 1671526523 | + |
| 3 | 1671526791 | - |
| 4 | 1671761020 | - |

| index | updated_timestamp | [n] >= [n+1] |
| --- | --- | --- |
| 0 | 1671770892 | |
| 1 | 1671684496 | + |
| 2 | 1671526523 | + |
| 3 | 1671526791 | - |
| 4 | 1671761020 | - |

**Comment by openapi-support** [comment]
Thank you for reporting this issue. We can confirm that sorting by updated timestamp (`sort_on=updated`) is currently not supported in our API, despite being documented as available.

* **Current Status**: We're investigating the implementation of this feature. Our technical analysis has identified that this requires infrastructure updates to support this sorting capability.
* **Technical Context**: While our API accepts the `sort_on=updated` parameter, the feature is not fully implemented. This explains why requests fall back to creation date sorting instead.
* **Next Steps**: We're working with our engineering teams to determine the implementation timeline and technical requirements.

We'll update this issue with when the problem is resolved.
